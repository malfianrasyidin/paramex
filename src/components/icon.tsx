import classNames from 'classnames';
import { IconName } from '@fortawesome/fontawesome-common-types';

export enum IconSize {
  ExtraSmall = 'fa-xs',
  Small = 'fa-sm',
  Medium = 'fa-md',
  Large = 'fa-lg',
  TwoTimes = 'fa-2x',
  ThreeTimes = 'fa-3x',
  FiveTimes = 'fa-5x',
  SevenTimes = 'fa-7x',
  TenTimes = 'fa-10x',
}

export enum IconStyle {
  Solid = 'fas',
  Regular = 'far',
  Light = 'fal',
  DuoTone = 'fad',
  Brands = 'fab',
}

export interface IconProps {
  name: IconName;
  size?: IconSize;
  style?: IconStyle;
  className?: string;
  onClick?: () => void;
  [x: string]: any;
}

export const Icon = ({
  name,
  size = IconSize.Medium,
  style = IconStyle.Solid,
  className,
  onClick,
  ...rest
}: IconProps) => {
  const args = Object.keys(rest).map((arg) => `fa-${arg}`);

  return (
    <i
      className={classNames(style, size, `fa-${name}`, args, className)}
      onClick={onClick}
    />
  );
};
