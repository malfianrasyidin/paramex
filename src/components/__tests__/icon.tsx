import { render } from '@testing-library/react';
import { Icon } from '../icon';

describe('Icon', () => {
  it('renders correctly', () => {
    const { container } = render(<Icon name="camera" />);

    expect(container).toMatchSnapshot();
  });
});
