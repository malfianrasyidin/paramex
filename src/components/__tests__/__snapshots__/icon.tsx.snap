// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders correctly 1`] = `
<div>
  <i
    class="fas fa-md fa-camera"
  />
</div>
`;
